### ISSUE TITLE
Error accessing OAI endpoint

### ISSUE THREAD

**Issue by IsaacHaze** [findsorguk/findsorguk#843]
I'm trying to harvest the finds.org.uk collection metadata via its OAI endpoint (http://www.finds.org.uk/datalabs/oai/request), but I get back a malformed (non) XML document after passing the first resumption token.

It seems that there is some trailing garbage:

```
            ...
            </metadata>
        </record>
        <resumptionToken expirationDate="2015-06-03T13:01:37Z" completeListSize="619930" cursor="60">216589</resumptionToken>
    </ListRecords>
</OAI-PMH>
<br />
<b>Notice</b>: Undefined index: creator in <b>/var/www/finds.org.uk/library/Pas/OaiPmhRepository/Metadata/OaiDc.php</b> on line <b>68</b><br />
<br />
<b>Notice</b>: Undefined index: creator in <b>/var/www/finds.org.uk/library/Pas/OaiPmhRepository/Metadata/OaiDc.php</b> on line <b>68</b><br />
<br />
<b>Notice</b>: Undefined index: creator in <b>/var/www/finds.org.uk/library/Pas/OaiPmhRepository/Metadata/OaiDc.php</b> on line <b>68</b><br />
<br />
<b>Notice</b>: Undefined index: creator in <b>/var/www/finds.org.uk/library/Pas/OaiPmhRepository/Metadata/OaiDc.php</b> on line <b>68</b><br />
```

First request:

    https://finds.org.uk/datalabs/oai/request?verb=ListRecords&metadataPrefix=oai_dc

which returns a proper resultset with a resumptionToken, so i issue the second request:

    https://finds.org.uk/datalabs/oai/request?verb=ListRecords&resumptionToken=216720

which produces the malformed content described above. Now, if I copy the resumptionToken from the second request and fire a third request everything is fine (and so are all requests afterwards.)

This seems to happen for the metadataFormats 'oai_dc', 'cdwalite' and 'pnds_dc'. The other formats are fine (for the first ten or so resultsets): 'lido' and 'ese'.

**Comment by portableant** [comment]
Super thank you, on the case.

**Comment by portableant** [comment]
Isaac, can you see if that is fixed now?

**Comment by IsaacHaze** [comment]
Almost, but it gives a similar error after fetching about 11k records:

```
        </oai_dc:dc>
      </metadata>
    </record>
    <resumptionToken expirationDate="2015-06-06T07:29:04Z" completeListSize="620359" cursor="5600">217376</resumptionToken>
  </ListRecords>
</OAI-PMH>
<br />
<b>Notice</b>:  Undefined index: imagedir in <b>/var/www/finds.org.uk/library/Pas/OaiPmhRepository/Metadata/OaiDc.php</b> on line <b>87</b><br />
<br />
<b>Notice</b>:  Undefined index: imagedir in <b>/var/www/finds.org.uk/library/Pas/OaiPmhRepository/Metadata/OaiDc.php</b> on line <b>87</b><br />
<br />
<b>Notice</b>:  Undefined index: imagedir in <b>/var/www/finds.org.uk/library/Pas/OaiPmhRepository/Metadata/OaiDc.php</b> on line <b>87</b><br />
<br />
<b>Notice</b>:  Undefined index: imagedir in <b>/var/www/finds.org.uk/library/Pas/OaiPmhRepository/Metadata/OaiDc.php</b> on line <b>87</b><br />
<br />
<b>Notice</b>:  Undefined index: imagedir in <b>/var/www/finds.org.uk/library/Pas/OaiPmhRepository/Metadata/OaiDc.php</b> on line <b>87</b><br />
<br />
<b>Notice</b>:  Undefined index: imagedir in <b>/var/www/finds.org.uk/library/Pas/OaiPmhRepository/Metadata/OaiDc.php</b> on line <b>87</b><br />
```

**Comment by portableant** [comment]
You didn't happen to save the output from that file? Just wondering who the person might be that is throwing that error. Each person has an image directory normally, must be a rogue....

**Comment by IsaacHaze** [comment]
Copy-n-pasted from my log file, see: https://gist.github.com/IsaacHaze/47825ddb4b70d5c581c1

**Comment by portableant** [comment]
Thanks. At a conference today, will fix when I can.

**Comment by portableant** [comment]
I am hoping this is fixed. Code is honky and needs rewriting, but I don't have enough time at the moment. I have increased the results returned for each call to 100...

**Comment by IsaacHaze** [comment]
Seems to work, i fetched 1240990 records without any error. Thanks for fixing.

**Comment by portableant** [comment]
Super, running tests with oaiharvester for r at the moment.

**Comment by portableant** [comment]
Closed by af75aa468c9a8a2216c6ac677344ef128c88a04e
